src/XrdHttp/XrdHttpReq.cc

```diff
@@ -1292,22 +1296,25 @@ int XrdHttpReq::ProcessHTTPReq() {
             if ((offs >= filesize) || (offs+l > filesize)) {
               TRACE(ALL, " Requested range " << l << "@" << offs <<
               " is past the end of file (" << filesize << ")");
-              //prot->SendSimpleResp(522, NULL, NULL, (char *) "Invalid range request", 0);
-              return -1;
+              httpStatusCode = 552;
+              httpStatusText = "Invalid range request";
+              return sendFooterError("Requested range is past end of file");
             }
             
             if (!prot->Bridge->Run((char *) &xrdreq, 0, 0)) {
-              prot->SendSimpleResp(404, NULL, NULL, (char *) "Could not run read request.", 0, false);
-              return -1;
+              TRACEI(REQ, " Failed to run read request on the bridge.");
+              XrdHttpReq::mapXrdErrorToHttpStatus();
```

Comment:
Maybe you can put the error message in a string allowing you to not repeat the same string in the two calls?

Reply:
Done.